Tests/Unit/LoRaTools/MacCommandTests.cs

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.Tests.Unit.LoRaTools
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using global::LoRaTools;
+    using global::LoRaTools.Mac;
+    using global::LoRaTools.Regions;
+    using LoRaWan.Tests.Common;
+    using Newtonsoft.Json;
+    using Xunit;
+
+    public abstract class MacCommandTests
+    {
+        /// <summary>
+        /// Gets the CID of the MAC command.
+        /// </summary>
+        public abstract Cid Cid { get; }
```

Comment:
These properties should use the `protected` modifier. Also, I don't think they need to be abstract. If `MacCommandTest` has a constructor to take the values of these properties as inputs, which would be the same enforcement as making them abstract here, then the properties could just be non-virtual accessors.

Reply:
c676cd90bd84fc9909b847803d7bd7b9662946d8